src/java/org/apache/cassandra/net/LatencySubscribers.java

```diff
@@ -64,12 +64,14 @@ public void add(InetAddressAndPort address, long latency, TimeUnit unit)
     /**
      * Track latency information for the dynamic snitch
      *
-     * @param cb      the callback associated with this message -- this lets us know if it's a message type we're interested in
-     * @param address the host that replied to the message
+     * @param cb           the callback associated with this message -- this lets us know if it's a message type we're interested in
+     * @param responseVerb the verb to respond the request
+     * @param address      the host that replied to the message
+     * @param isTimeout    if the request has timed out
      */
-    public void maybeAdd(RequestCallback cb, InetAddressAndPort address, long latency, TimeUnit unit)
+    public void maybeAdd(RequestCallback cb, Verb responseVerb, InetAddressAndPort address, long latency, TimeUnit unit, boolean isTimeout)
     {
-        if (cb.trackLatencyForSnitch())
+        if (cb.trackLatencyForSnitch(responseVerb, isTimeout))
```

Comment:
Do we actually have to pass the response verb? Can't we use the `WriteType` which is already available in the response handler and check it's `COUNTER`?